src/reader.rs

```diff
@@ -6511,10 +6511,22 @@ impl<'a> Reader<'a> {
             prefix.push_utfstr(&tok);
             prefix.push_utfstr(&common_prefix);
         } else {
-            // Append just the end of the string.
+            // Collapse parent directories and append end of string
             prefix.push(get_ellipsis_char());
+
             let full = tok + &common_prefix[..];
-            prefix.push_utfstr(&full[full.len() - PREFIX_MAX_LEN..]);
+            let truncated = &full[full.len() - PREFIX_MAX_LEN..];
+            let parts: Vec<&wstr> = truncated.split('/').collect();
+            if parts.len() < 2 {
```

Comment:
`parts.len() < 2` is a bit confusing because there's always at least one part.
Better make that obvious by checking `parts.len() == 1`. 

Actually we should also get rid of the gratuitous vector, how about


	    ```rust
            let (i, last_component) = truncated.split('/').enumerate().last().unwrap();
            if i == 0 {
                // No path separators were found in the common prefix, so we can't collapse
                // any further
                prefix.push_utfstr(&truncated);
            } else {
                // Discard any parent directories and include whats left
                prefix.push('/');
                prefix.push_utfstr(last_component);
            };
	    ```